### ISSUE TITLE
How to send multiple responses(chunks) from Graphql server to client

### ISSUE THREAD

**Issue by vtahlani** [graphql/graphql-js#973]
How can I send multiple responses from graphql server to a client for its single request.
{
    users():{
        name
           friends():{
                name
           }
   }
}

In above schema, I send a request to users and friends services, here users service sends multiple partial responses(chunks) and as soon as i get first response(chunk) from users service i want to send a request to friends service while other responses(chunks) are coming from users service, once get response friends then i want to send first response back to client while server is still getting other chunks from users and friends service then send other chunks to client. If i receive N chunks from users service then i want to send N chunks to client as well.

Basically i want three task 1) sending request to users 2) sending request to friends after receiving response from users and sending response back to client work in parallel to avoid high memory utilization.

**chunk** - lets say i should get 1 millions as complete response so instead of sending everything in one shot server is going to send 10000 records as one response another 10000 records in next and so on........

**Comment by leebyron** [comment]
There is currently no way to do this outside of performing a batched query: many smaller queries sent collectively as a single request, where each query fulfilling send back as a chunk.

However streaming parts of a query is something we've been experimenting with for a long time - it turns out to be more difficult from the product consumption side than the server generation side. There is some ongoing work in Relay testing out these ideas.
